File: app/src/main/java/org/schabi/newpipe/database/LocalItem.java
package org.schabi.newpipe.database;

public interface LocalItem {
    LocalItemType getLocalItemType();

    enum LocalItemType {
        PLAYLIST_LOCAL_ITEM,
        PLAYLIST_REMOTE_ITEM,

        PLAYLIST_STREAM_ITEM,
        STATISTIC_STREAM_ITEM,
    }
}

File: app/src/main/java/org/schabi/newpipe/database/LocalItem.kt
/*
 * SPDX-FileCopyrightText: 2018-2020 NewPipe contributors <https://newpipe.net>
 * SPDX-FileCopyrightText: 2025 NewPipe e.V. <https://newpipe-ev.de>
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package org.schabi.newpipe.database

interface LocalItem {
    val localItemType: LocalItemType

    enum class LocalItemType {
        PLAYLIST_LOCAL_ITEM,
        PLAYLIST_REMOTE_ITEM,

        PLAYLIST_STREAM_ITEM,
        STATISTIC_STREAM_ITEM
    }
}
